fdb-relational-api/src/main/java/com/apple/foundationdb/relational/api/exceptions/ErrorCode.java

```diff
@@ -166,7 +166,8 @@ public enum ErrorCode {
      * Used for the exceptions due to internal issue of Relational, which is caused by neither clients nor upstream
      * system.
      */
-    INTERNAL_ERROR("XX000");
+    INTERNAL_ERROR("XX000"),
+    DESERIALIZATION_FAILURE("XXF01");
```

Comment:
Is this an internal error? 
If the wrong keys are passed, this sql state would be thrown, and that seems to be user driven rather than internal.

Should we use `40001` instead? 
If we don't want `40001` because it's `serialization_failure`, I would suggest a new `40F01` `deserialization_failure`.

Reply:
`SERIALIZATION_FAILURE` is not SerDe; it is serializable isolation.

I am not sure how incorrect key can be distinguished from data corruption, which other vendors tend to treat as an internal error.